### PR TITLE
Always return Promise from getFeatures

### LINKED ISSUE #12065: getFeatures returns error when the layer is added but not yet rendered
**Describe the bug**
Calling getFeatures on a `ol/layer/VectorTile` that is not yet rendered but visible throws an error because this.renderer_ is not yet set.
https://github.com/openlayers/openlayers/blob/b6e7d617b5860aba7dd4dc8835085c415c64d731/src/ol/layer/Layer.js#L212-L214
This likely also happens with `ol/layer/Vector`.

**To Reproduce**
1. Go to https://codesandbox.io/s/getfeatures-error-after-load-qs14b?file=/main.js
2. After page is loaded move the mouse over the map until the layer is added
3. When the layer is added 3 seconds later you get an error

**Expected behavior**
There should be no error, it should always return a Promise.

### PR DESCRIPTION
**Pull request by M393**
Fixes #12065
Also changed `VectorImageLayer`'s getFeatures function to follow the same pattern, no functional change there.

### CODE REVIEW

**Review by ahocevar — APPROVED** [review]
Thanks, @M393

File: src/ol/layer/Layer.js [hunk 1]
```diff
@@ -210,6 +210,9 @@ class Layer extends BaseLayer {
    * an array of features.
    */
   getFeatures(pixel) {
+    if (!this.renderer_) {
+      return new Promise((resolve) => resolve([]));
```
Comment (tschaub) [inline src/ol/layer/Layer.js:214]: Minor, but this could be `return Promise.resolve([])`.

Comment (M393) [reply src/ol/layer/Layer.js:214]: Thanks @tschaub, I did not know this. Should I create a PR?

Comment (tschaub) [reply src/ol/layer/Layer.js:214]: Not critical.

**Pull request merged**